### PR TITLE
54394 - adding required fields helper functions, in general-template.php

### PR DESCRIPTION
**Pull request by sabernhardt**
In this PR, the helper functions of [PR 2821](https://github.com/WordPress/wordpress-develop/pull/2821) are simplified and moved to general-template.php.

This also adds a space before the indicator glyph in `get_media_item()` and `get_compat_media_markup()` that was not there before, for consistency with the other functions.

Trac ticket: https://core.trac.wordpress.org/ticket/54394

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/general-template.php [hunk 1]
```diff
@@ -4906,6 +4906,38 @@ function __checked_selected_helper( $helper, $current, $echo, $type ) { // phpcs
...
 
+/**
+ * Assigns a visual indicator for required form fields.
+ *
+ * @since 6.1.0
+ *
+ * @return string Indicator glyph wrapped in a `span` tag.
+ */
+function wp_required_field_indicator() {
+	/* translators: Character to identify required form fields. */
```
Comment (audrasjb) [inline src/wp-includes/general-template.php:4918]: maybe we could have something like this?
```
/* translators: Character to identify required form fields. */
$indicator = '<span class="required" aria-hidden="true">' . esc_html__( '*' ) . '</span>';
```
Not sure about that myself, but worth asking :)

**Review by audrasjb — COMMENTED** [review]
Another small nitpick. Not sure about this myself :)

File: src/wp-includes/general-template.php (hunk 1 above)
Comment (sabernhardt) [reply src/wp-includes/general-template.php:4918]: This makes sense on its own, but I had them on separate lines to prepare for the filter/argument proposal on [Trac 23870](https://core.trac.wordpress.org/ticket/23870).

Comment (audrasjb) [reply src/wp-includes/general-template.php:4918]: I see, thanks :)

**Comment by audrasjb** [comment]
Committed in https://core.trac.wordpress.org/changeset/53888

**Pull request closed without merging**